### ISSUE TITLE
Support git submodule filters

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#33157]
### Discussed in https://github.com/renovatebot/renovate/discussions/33027

<div type='discussions-op-text'>

<sup>Originally posted by **Kobzol** December 10, 2024</sup>
### Tell us more.

Hi! I'm a member of the Rust infra team, and I'd like to get renovatebot working on the https://github.com/rust-lang/rust repository, which hosts the Rust toolchain (compiler, standard library etc.), using the Forking Renovatebot GitHub app.

Currently, I'm facing two blockers:
- We need to use nightly cargo (this is already tracked in https://github.com/renovatebot/renovate/discussions/15728 and other issues)
- Slow speed of git submodules checkout

For updating certain Rust lockfiles, we need to checkout submodules in the Rust repository. However, the repo has a lot of submodules, some of which are very large (e.g. LLVM or GCC). I tried to set `"cloneSubmodules": true`, but then the renovatebot job times out after thirty minutes.

I wonder if it would be possible to either:
- Allow selecting which submodules will be checked out. We actually need only a few small-ish submodules for the lockfile updates, but currently Renovatebot either checks out all submodules or none.
- Make submodule checkouts faster, by using some git magic. For example, in the Rust repo, we have a [custom script](https://github.com/rust-lang/rust/blob/04acc1d9dcb57b973e818ea817b106af5214164f/src/ci/scripts/checkout-submodules.sh) that makes the checkouts faster by downloading archives from GitHub (although this is not a general solution, of course).

Thanks!</div>

**Comment by rarkins** [comment]
FYI @Kobzol

**Comment by rarkins** [comment]
Closed by https://github.com/renovatebot/renovate/pull/33115

Waiting for the next update of the Mend app and it will be available

**Comment by Kobzol** [comment]
Thank you for implementing this! Is there a way to find out what version is currently deployed in the Mend Renovate GitHub app? I haven't found it anywhere, and it doesn't seem to be deployed yet:
![image](https://github.com/user-attachments/assets/3037d882-ccf5-4d25-9976-65295d7bb9bd)

**Comment by rarkins** [comment]
See https://docs.renovatebot.com/mend-hosted/hosted-apps-config/#renovate-version

**Comment by Kobzol** [comment]
Thanks! I tried to search for it in the log, but it was a bit later than I expected.

**Comment by Kobzol** [comment]
I can confirm that this works and it has helped unblock Renovatebot submodule support in `rust-lang/rust` :tada: Thank you very much! Now only the nightly Cargo issue remains.
